File: 1886-determine-whether-matrix-can-be-obtained-by-rotation/1886-determine-whether-matrix-can-be-obtained-by-rotation.py
class Solution:
    def findRotation(self, mat: List[List[int]], target: List[List[int]]) -> bool:
        if mat == target:
            return True
        col = len(mat)
        row = len(mat[0])

        rotate_mat = []
        for c in range(row):
            new_row = []

            for r in range(col):
                new_row.append(mat[r][c])
            print(new_row[::-1])
            rotate_mat.append(new_row[::-1])

        if rotate_mat == target:
            return True
        else:
            mat = rotate_mat
            rotate_mat = []
            for c in range(row):
                new_row = []

                for r in range(col):
                    new_row.append(mat[r][c])
                print(new_row[::-1])
                rotate_mat.append(new_row[::-1])

            print(rotate_mat, target)

            if rotate_mat == target:
                return True
            else:
                mat = rotate_mat
                rotate_mat = []
                for c in range(row):
                    new_row = []

                    for r in range(col):
                        new_row.append(mat[r][c])
                    print(new_row[::-1])
                    rotate_mat.append(new_row[::-1])

                print(rotate_mat, target)

                if rotate_mat == target:
                    return True
        
        return False

File: 1886-determine-whether-matrix-can-be-obtained-by-rotation/1886-determine-whether-matrix-can-be-obtained-by-rotation.go
func compare(mat1, mat2 [][]int) bool {
	if len(mat1) != len(mat2) || len(mat1[0]) != len(mat2[0]) {
		return false
	}
	for i := range mat1 {
		for j := range mat1[i] {
			if mat1[i][j] != mat2[i][j] {
				return false
			}
		}
	}
	return true
}

func rotate(mat [][]int) [][]int {
	n := len(mat)
	rotated := make([][]int, n)
	for i := range rotated {
		rotated[i] = make([]int, n)
	}
	for i := 0; i < n; i++ {
		for j := 0; j < n; j++ {
			rotated[j][n-1-i] = mat[i][j]
		}
	}
	return rotated
}

func findRotation(mat [][]int, target [][]int) bool {
	for i := 0; i < 4; i++ {
		if compare(mat, target) {
			return true
		}
		mat = rotate(mat)
	}
	return false
}
